src/components/ToDoApp.js

```diff
@@ -0,0 +1,281 @@
+   */
+  handleOnSubmit = (e) => {
+    e.preventDefault();
+    const submittedText = this.state.currentText;
+    const newToDoObj = {
+      todoContent: submittedText,
+      doneStatus: false,
+      editingStatus: false,
+      id: Date.now().toString()
+    };
+    const newToDoList = [newToDoObj, ...this.state.todoList];
+
+    this.setState({
+      todoList: newToDoList,
+      currentText: ''
+    });
+  }
+
+  /**
+   * Handles the checbox checked/unchecked states. Passes the event 'e' to handleOnDone function.
+   *
+   * @param {Event} e
+   * @memberof ToDoApp
+   */
+  handleCheckBoxChange = (e) => {
+    this.handleOnDone(e);
+  }
+
+  /**
+   * Handles changing values in the edit input field when in editStatus is true.
+   *
+   * @param {Event} e
+   * @memberof ToDoApp
+   */
+  handleEditChange = (e) => {
+    const editingId = e.target.parentNode.parentNode.getAttribute('id');
+    const newToDoList = [].concat(this.state.todoList);
+    const editingObj = Object.assign({}, newToDoList.filter((item) => {
+      return item.id === editingId;
+    })[0]);
+
+    editingObj.todoContent = e.target.value;
+    newToDoList[newToDoList.findIndex((item) => {
+      return item.id === editingId;
+    })] = editingObj;
+    this.setState({
+      todoList: newToDoList
+    });
+  }
+
+
+  /**
+   * Handles setting the todo item's new value when submitted on edit input field.
+   *
+   * @param {Event} e
+   * @memberof ToDoApp
+   */
+  onEditSubmit = (e) => {
+    e.preventDefault();
+    const editingId = e.target.parentNode.getAttribute('id');
+    const newToDoList = [].concat(this.state.todoList);
+    const editingObj = Object.assign({}, newToDoList.filter((item) => {
+      return item.id === editingId;
+    })[0]);
+    const formChilds = e.target.childNodes;
+
+    editingObj.todoContent = formChilds[0].value;
+    editingObj.editingStatus = false;
+    newToDoList[newToDoList.findIndex((item) => {
+      return item.id === editingId;
+    })] = editingObj;
+    this.setState({
+      todoList: newToDoList
+    });
+  }
+
+  /**
+   * Filters the list to be displayed according to currentList(all,completed, remaining) value.
+   *
+   * @returns {Array}
+   * @memberof ToDoApp
+   */
+  filterDisplayList = () => {
+    switch(this.state.currentList) {
+      case TABS.all:
+        return this.state.todoList;
+      case TABS.completed:
+        return this.state.todoList.filter((todoItem) => {
+          return todoItem.doneStatus;
+        });
+      case TABS.remaining:
+        return this.state.todoList.filter((todoItem) => {
+          return !todoItem.doneStatus;
+        });
+      default:
+        return this.state.todoList;
+    }
+  }
+
+  /**
+   * Handles changing currentList's value when tabs on the navigation are clicked.
+   *
+   * @param {String} tab Value of pressed tab (all, completed, remaining).
+   * @param {Event} e
+   * @memberof ToDoApp
+   */
+  setCurrentTab = (tab, e) => {
```

Comment:
it's a good practice to have an event as the first parameter so that you can extend other params